PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'mrp.production'
-    _inherit = 'product.product'
-    _inherit = 'stock.move'
-    _inherit = 'stock.picking'

Total Changes: 9

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/mrp_subcontracting_account/models/mrp_production.py
+++ b/addons/mrp_subcontracting_account/models/mrp_production.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class MrpProduction(models.Model):
-    _inherit = 'mrp.production'
+    _inherit = ['mrp.production']
 
     def _cal_price(self, consumed_moves):
         finished_move = self.move_finished_ids.filtered(lambda x: x.product_id == self.product_id and x.state not in ('done', 'cancel') and x.quantity > 0)

--- a/addons/mrp_subcontracting_account/models/product_product.py
+++ b/addons/mrp_subcontracting_account/models/product_product.py
@@ -3,8 +3,9 @@
 
 from odoo import models, fields
 
+
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     def _compute_bom_price(self, bom, boms_to_recompute=False, byproduct_bom=False):
         """ Add the price of the subcontracting supplier if it exists with the bom configuration.

--- a/addons/mrp_subcontracting_account/models/stock_move.py
+++ b/addons/mrp_subcontracting_account/models/stock_move.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockMove(models.Model):
-    _inherit = 'stock.move'
+    _inherit = ['stock.move']
 
     def _should_force_price_unit(self):
         self.ensure_one()

--- a/addons/mrp_subcontracting_account/models/stock_picking.py
+++ b/addons/mrp_subcontracting_account/models/stock_picking.py
@@ -6,7 +6,7 @@ from odoo.osv.expression import OR
 
 
 class StockPicking(models.Model):
-    _inherit = 'stock.picking'
+    _inherit = ['stock.picking']
 
     def action_view_stock_valuation_layers(self):
         action = super(StockPicking, self).action_view_stock_valuation_layers()
